avogadro/qtgui/pluginlayermanager.h

```diff
@@ -28,7 +28,8 @@ class AVOGADROQTGUI_EXPORT PluginLayerManager : protected Core::LayerManager
   ~PluginLayerManager();
 
   /** @return if the active layer in the molecule is locked. */
-  static bool activeLayerLocked();
+  bool activeLayerLocked() const;
+  bool atomLocked(size_t atom) const;
```

Comment:
This might also be useful in the `QtGui::Molecule` class. I'm working on dynamics code and I want to know which atoms are locked / frozen so I don't move them.

Reply:
That could be interesting, but maybe you are looking for `LayerManager::getMoleculeInfo(mol)` cause the `pluginLayer` only works with the active molecule, or a custom version from it.